pkg/errors/errors.go

```diff
@@ -0,0 +1,143 @@
+}
+
+// Format implements the fmt.Formatter interface to support the formatting of an error chain with "%+v" verb.
+func (b *base) Format(s fmt.State, verb rune) {
+	if verb == 'v' && s.Flag('+') {
+		s.Write([]byte(formatErrorChain(b)))
+		return
+	}
+
+	s.Write([]byte(b.Error()))
+}
+
+// New creates a new error with the given message and a stacktrace in details.
+// An alternative to errors.New function.
+func New(msg string) error {
+	return &base{
+		info:  msg,
+		stack: newStackTrace(),
+		err:   nil,
+	}
+}
+
+// Errorf creates a new error with the given message and a stacktrace in details.
+// An alternative to fmt.Errorf function.
+func Errorf(format string, args ...interface{}) error {
+	return &base{
+		info:  fmt.Sprintf(format, args...),
+		stack: newStackTrace(),
+		err:   nil,
+	}
+}
+
+// Wrap creates a new error with the given message, wrapping another error and a stacktrace in details.
+// If cause is nil, this is the same as New.
+func Wrap(cause error, msg string) error {
+	return &base{
+		info:  msg,
+		stack: newStackTrace(),
+		err:   cause,
+	}
+}
+
+// Wrapf creates a new error with the given message, wrapping another error and a stacktrace in details.
+// If cause is nil, this is the same as Errorf.
+func Wrapf(cause error, format string, args ...interface{}) error {
+	return &base{
+		info:  fmt.Sprintf(format, args...),
+		stack: newStackTrace(),
+		err:   cause,
+	}
+}
+
+// Unwrap is a wrapper of built-in errors.Unwrap. Unwrap returns the result of
+// calling the Unwrap method on err, if err's type contains an Unwrap method
+// returning error. Otherwise, Unwrap returns nil.
+func Unwrap(err error) error {
+	return errors.Unwrap(err)
+}
+
+// UnwrapTillCause returns the result of calling the Unwrap method on err, if err's
+// type contains an Unwrap method returning error. Otherwise, UnwrapTillCause returns
+// the last encountered error. The difference between Unwrap and UnwrapTillCause is the
+// first one returns nil if the error does not implement interface with Unwrap method but
+// UnwrapTillCause returns the last err (whether it's nil or not) where it failed to assert
+// the interface.
+// This is a replacement of errors.Cause without the causer interface from pkg/errors which
+// actually can be sufficed through the errors.Is function. But considering some use cases
+// where we need to peel off all the external layers applied through errors.Wrap family,
+// it is useful ( where external SDK doesn't use errors.Is internally).
+func UnwrapTillCause(err error) error {
+	u, ok := err.(interface {
+		Unwrap() error
+	})
+	if !ok {
+		// here is the difference, Unwrap returns nil here
+		return err
+	}
+	return u.Unwrap()
+}
+
+// Is is a wrapper of built-in errors.Is. It reports whether any error in err's
+// chain matches target. The chain consists of err itself followed by the sequence
+// of errors obtained by repeatedly calling Unwrap.
+func Is(err, target error) bool {
+	return errors.Is(err, target)
+}
```

Comment:
wonder if alias `var Is = errors.Is` would not work? (and same for `As`).

Reply:
It would work, updating

Reply:
Hey, sorry, I have reverted back to the old one.
Reason: Though using `var Is = errors.Is` would definitely make things work but users will have a slightly poor IDE experience as the ide will treat those functions as variables (as actually, they are in the modified pkg). So the auto-completion works a bit weird way
see the screenshot
![image](https://user-images.githubusercontent.com/41498427/162378927-be520b63-8e3b-45e1-b614-e3a85806aa8e.png)

I dug in a little bit to find an alternative approach to tackle this, but it seems popular packages do use the wrapping while exposing their client-side APIs from the internal packages.
ref: https://github.com/temporalio/sdk-go/blob/fd0d1eb548eb0621a5395581cfe2c418704b007c/client/client.go#L435-L476

I hope you are okay with it. @bwplotka?

Reply:
Makes sense, but worth adding comment why we did this this way 🙃

Reply:
Sure : )

Reply:
BTW I know why it did not work for you. We can do `type Is = errors.Is` I think. Let's merge and iterate (:

Reply:
> I think this needs rebasing, the docs check will keep failing due to an outdated link in the docs disappointed cc @bisakhmondal

Done. Thanks to you both for the awesome review : )